cli/cmd/install_test.go

```diff
@@ -318,25 +318,31 @@ metadata:
 }
 
 func TestRenderCRDsWithMissingGatewayAPI(t *testing.T) {
-
 	k, err := k8s.NewFakeAPIFromManifests([]io.Reader{})
 	if err != nil {
 		t.Fatalf("failed to initialize fake API: %s", err)
 	}
 
 	var buf bytes.Buffer
-	err = renderCRDs(context.Background(), k, &buf, values.Options{}, "yaml")
+	err = renderCRDs(context.Background(), k, &buf, values.Options{
+		Values: []string{"installGatewayAPI=false"},
+	}, "yaml")
 	if err == nil {
 		t.Fatalf("Installing with missing Gateway API CRDs should fail")
 	}
 }
 
 func TestRenderCRDsWithGatewayAPI(t *testing.T) {
+	k, err := k8s.NewFakeAPIFromManifests([]io.Reader{})
+	if err != nil {
+		t.Fatalf("failed to initialize fake API: %s", err)
+	}
+
 	options := values.Options{
 		Values: []string{"installGatewayAPI=true"},
 	}
 	var buf bytes.Buffer
-	if err := renderCRDs(context.Background(), nil, &buf, options, "yaml"); err != nil {
+	if err := renderCRDs(context.Background(), k, &buf, options, "yaml"); err != nil {
```

Comment:
The fact that the API was not passed into the function caused the whole check that is defined here https://github.com/linkerd/linkerd2/blob/16b74894f9978cafd0f9e17c3a0ca1386358d4e8/cli/cmd/install.go#L406 to be missed, rendering this test a no-op

Reply:
very good catch